### ISSUE TITLE
SA1206 does not correctly handle the required keyword ordering

### ISSUE THREAD

**Issue by ThiagoDSMarcelino** [DotNetAnalyzers/StyleCopAnalyzers#3928]
According to the [SA1206 documentation](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1206.md), keywords should appear in the following order:

1. **Access modifiers**
2. `static`
3. All other keywords

However, when using the `required` keyword, the analyzer incorrectly reports a warning:

```csharp
public sealed class Foo
{
    public required string Bar { get; set; }
}

// Warning: "The 'required' modifier should appear before 'public'"
```

This behavior seems inconsistent with the documented ordering rules. The required keyword should fall under "All other keywords," appearing after the access modifier.

Could you please clarify whether this is an issue in the analyzer or a documentation gap regarding the ordering of `required`?

**Comment by slaneyrw** [comment]
Do we have a timeframe for non-beta release.  Local policy does not allow beta packages.
